enterprise/backend/src/metabase_enterprise/remote_sync/api.clj

```diff
@@ -0,0 +1,279 @@
+(ns metabase-enterprise.remote-sync.api
+  (:require
+   [clojure.string :as str]
+   [diehard.core :as dh]
+   [medley.core :as m]
+   [metabase-enterprise.remote-sync.impl :as impl]
+   [metabase-enterprise.remote-sync.models.remote-sync-object :as remote-sync.object]
+   [metabase-enterprise.remote-sync.models.remote-sync-task :as remote-sync.task]
+   [metabase-enterprise.remote-sync.settings :as settings]
+   [metabase-enterprise.remote-sync.source :as source]
+   [metabase-enterprise.remote-sync.source.ingestable :as source.ingestable]
+   [metabase-enterprise.remote-sync.source.protocol :as source.p]
+   [metabase.api.common :as api]
+   [metabase.api.macros :as api.macros]
+   [metabase.api.routes.common :refer [+auth]]
+   [metabase.app-db.cluster-lock :as cluster-lock]
+   [metabase.collections.models.collection :as collection]
+   [metabase.util.jvm :as u.jvm]
+   [metabase.util.log :as log]
+   [metabase.util.malli.schema :as ms]))
+
+(set! *warn-on-reflection* true)
+
+(defn- run-async!
+  [ttype branch f]
+  (let [{task-id :id
+         existing? :existing?}
+        (cluster-lock/with-cluster-lock impl/cluster-lock
+          (if-let [{id :id} (remote-sync.task/current-task)]
+            {:existing? true :id id}
+            (remote-sync.task/create-sync-task! ttype api/*current-user-id*)))]
+    (api/check-400 (not existing?) "Remote sync in progress")
+    (u.jvm/in-virtual-thread*
+     (dh/with-timeout {:interrupt? true
```

Comment:
Maybe it would be better to make the timeout here a multiple of the remote-sync-task-time-limit-ms instead? I'd like this setting to default to a lower value now that progress is regularly reported so this is a safety-valve and doesn't need to following this setting exactly.